src/eduid_userdb/reset_password/state.py

```diff
@@ -209,15 +213,15 @@ def __init__(
         self._data_in = copy.deepcopy(cast(dict, data))  # to not modify callers data
         self._data = dict()
 
-        email_address = self._data_in.pop('email_address')
-        email_code = self._data_in.pop('email_code')
+        mail_address: str = self._data_in.pop('email_address')
```

Comment:
Where does this certainty that data['email_address'] is a string stem from? I don't see any guarantees that it is a string, and in case we don't know it is a string, we throw away the value of type checking by incorrectly labeling it a string.

Maybe this object is 'leaf enough' that we can just throw away the 'data' parameter from the constructor and change it into a from_dict right away?

Reply:
This is definitely a string, at least the way it is used in the webapp code. Objects of this class [are created from an email address string just received](https://github.com/SUNET/eduid-webapp/blob/master/src/eduid_webapp/security/helpers.py#L156) from an anonymous user; all we do before creating the state is check that it corresponds to some user in our db. It is subsequently used as a string throughout the code.
And tbh I don't see much value in labelling the email for a password reset state as verified or primary - in the state. The address in our user db may have been verified by some user, but we have no guarantee that it is the same user that is requesting the password reset.

Reply:
What about if the data dict is used?

My preferred solution would be to get rid of the data dict option, but otherwise you need to add a runtime type check and not just promise the type checking system that what comes out of the dict is a string.